erizo/src/erizo/NicerConnection.cpp

```diff
@@ -320,6 +320,16 @@ void NicerConnection::startGathering() {
 }
 
 bool NicerConnection::setRemoteCandidates(const std::vector<CandidateInfo> &candidates, bool is_bundle) {
+  // At this moment we should have peer_ initalized. To make sure, we are waiting for start_promise_ there
```

Comment:
Shouldn't we make sure we don't call setRemoteCandidates before having started Nicer? I don't like the idea of blocking the thread while we are waiting. Another alternative is to return false and call setRemoteCandidates after a while again.